core/src/com/unciv/logic/multiplayer/DropBox.kt

```diff
@@ -0,0 +1,169 @@
+    }
+
+    fun downloadFile(fileName: String): InputStream {
+        val response = dropboxApi("https://content.dropboxapi.com/2/files/download",
+                contentType = "text/plain", dropboxApiArg = "{\"path\":\"$fileName\"}")
+        return response!!
+    }
+
+    fun downloadFileAsString(fileName: String): String {
+        val inputStream = downloadFile(fileName)
+        return BufferedReader(InputStreamReader(inputStream)).readText()
+    }
+
+    /**
+     * @param overwrite set to true to avoid DropBoxFileConflictException
+     * @throws DropBoxFileConflictException when overwrite is false and a file with the
+     * same name already exists
+     */
+    fun uploadFile(fileName: String, data: String, overwrite: Boolean = false) {
+        val overwriteModeString = if(!overwrite) "" else ""","mode":{".tag":"overwrite"}"""
+        dropboxApi("https://content.dropboxapi.com/2/files/upload",
+                data, "application/octet-stream", """{"path":"$fileName"$overwriteModeString}""")
+    }
+
+    fun deleteFile(fileName: String){
+        dropboxApi("https://api.dropboxapi.com/2/files/delete_v2",
+                "{\"path\":\"$fileName\"}", "application/json")
+    }
+
+    fun fileExists(fileName: String): Boolean {
+        try {
+            dropboxApi("https://api.dropboxapi.com/2/files/get_metadata",
+                    "{\"path\":\"$fileName\"}", "application/json")
+            return true
+        } catch (ex: FileNotFoundException) {
+            return false
+        }
+    }
+
+    fun getFileMetaData(fileName: String): IFileMetaData {
+        val stream = dropboxApi("https://api.dropboxapi.com/2/files/get_metadata",
+                "{\"path\":\"$fileName\"}", "application/json")!!
+        val reader = BufferedReader(InputStreamReader(stream))
+        return GameSaver.json().fromJson(DropboxMetaData::class.java, reader.readText())
+    }
+
+//
+//    fun createTemplate(): String {
+//        val result =  dropboxApi("https://api.dropboxapi.com/2/file_properties/templates/add_for_user",
+//                "{\"name\": \"Security\",\"description\": \"These properties describe how confidential this file or folder is.\",\"fields\": [{\"name\": \"Security Policy\",\"description\": \"This is the security policy of the file or folder described.\nPolicies can be Confidential, Public or Internal.\",\"type\": \"string\"}]}"
+//                ,"application/json")
+//        return BufferedReader(InputStreamReader(result)).readText()
+//    }
+
+    @Suppress("PropertyName")
+    class FolderList{
+        var entries = ArrayList<DropboxMetaData>()
+        var cursor = ""
+        var has_more = false
+    }
+
+    @Suppress("PropertyName")
+    class DropboxMetaData: IFileMetaData {
+        var name = ""
+        private var server_modified = ""
+
+        override fun getFileName() = name
+        override fun getLastModified(): Date {
+            return server_modified.parseDate()
+        }
+    }
```

Comment:
Merged FolderListEntry with DropboxMetaData as they were basically the same
Modified DropboxMetaData to use IFileMetaData